src/dbnode/storage/block/block.go

```diff
@@ -413,21 +413,34 @@ func (b *dbBlock) resetRetrievableWithLock(
 	b.wasRetrievedFromDisk = false
 }
 
+func (b *dbBlock) Discard() ts.Segment {
```

Comment:
Fix locking / pooling issue here

Reply:
I'm actually not sure there is a good solution here. We could maybe do something asynchronous maybe, but it would expensive / slower. This actually seems like the best solution and we handle the closing/pooling of the series object the same way.

Since this object will get placed into the "back" of the pool, it seems unlikely to cause performance issues, and any alternatives will probably be slower.

Reply:
Can you change it to this?